x/mongo/driver/crypt.go

```diff
@@ -399,7 +397,10 @@ func (c *crypt) decryptKey(kmsCtx *mongocrypt.KmsContext) error {
 
 		res := make([]byte, bytesNeeded)
 		bytesRead, err := conn.Read(res)
-		if err != nil && !errors.Is(err, io.EOF) {
+		if err != nil {
+			if kmsCtx.Fail() {
```

Comment:
See MONGOCRYPT-752. When `mongocrypt_kms_ctx_fail` returns false, consider wrapping the error message from `mongocrypt_ctx_kms_status`. The error from `mongocrypt_ctx_kms_status` may help identify a retry occurred (e.g. "KMS request failed after 3 retries due to a network error: last attempt failed with: <driver error>")